**<p style = "text-align: center">TRƯỜNG ĐẠI HỌC KHOA HỌC TỰ NHIÊN - ĐẠI HỌC QUỐC GIA TPHCM</p>**

# **<p style = "text-align: center">NHẬP MÔN KHOA HỌC DỮ LIỆU</p>**
## **<p style = "text-align: center">Đồ án Cuối kì</p>**
### <p style = "text-align: center"> <b>Đề tài:</b> Phân tích, khám phá dữ liệu về sách trên web thương mại điện tử (tiki)</p>
**Sinh viên thực hiện**
| Mã số sinh viên | Họ và tên |
| --- | --- |
| 20120040 | Nguyễn Quang Gia Bảo |
| 20120136 | Huỳnh Tuấn Nam |
| 20120158 | Trần Hoàng Anh Phi |

---

# **1. Thông tin về dữ liệu**<br>
Dữ liệu được sử dụng trong bài được lấy từ trang: <a href = "https://tiki.vn/sach-truyen-tieng-viet/c316"> https://tiki.vn/sach-truyen-tieng-viet/c316</a><br>

Vì website <a href = 'https://tiki.vn/'> https://tiki.vn/ </a> có khá là nhiều mặt hàng nên nhóm chỉ chọn ra danh mục <font color=red>sách tiếng Việt</font> để crawl và thực hiện đồ án này.


# **2. Khám phá dữ liệu**<br>

#### Import các package cần thiết

In [2]:
import pandas as pd


#### Đọc dữ liệu đã thu thập từ file csv 

In [27]:
df = pd.read_csv('full_data/data.csv')
print(df.shape)
display(df.head())

C:\Users\Admin\AppData\Local\Temp\ipykernel_18800\906639567.py:1: DtypeWarning: Columns (22,27,29,30,31,32,33,34,35,36,37,38,39,40,41,43,45,46,47,48,49,50,51,52,53,54,56,57,58,59,60,61,62,63,64,65,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('full_data/data.csv')


(96647, 84)


Unnamed: 0         id  master_id            sku  \
0           0  204317934  204317934  9972713357587   
1           1  205279963  205279963  7618382614299   
2           2  193209826  193209826  5001141446127   
3           3  205288789  205288789  9035774965791   
4           4  204649516  204649516  8708257676679   

                                                name  \
0     1111 - Nhật Ký Sáu Vạn Dặm Trên Yên Xe Cà Tàng   
1     1111 - Nhật Ký Sáu Vạn Dặm Trên Yên Xe Cà Tàng   
2  Con Đường Tơ Lụa - Vạn Dặm Xa Từ Pakistan Tới ...   
3     1111 - Nhật Ký Sáu Vạn Dặm Trên Yên Xe Cà Tàng   
4  Sách - 1111 - NHẬT KÝ SÁU VẠN DẶM TRÊN YÊN XE ...   

                                           short_url book_cover   price  \
0  https://tiki.vn/product-p204317934.html?spid=2...        NaN  262000   
1  https://tiki.vn/product-p205279963.html?spid=2...    Bìa mềm  256750   
2  https://tiki.vn/product-p193209826.html?spid=1...    Bìa mềm   95100   
3  https://tiki.vn/product-p205288789.html?spid=2...    Bìa mềm  292500   
4  https://tiki.vn/product-p204649516.html?spid=2...        NaN  292500   

   original_price  discount_rate  ...  kich_thuoc  tai_trong device_brand  \
0          325000             19  ...         NaN        NaN          NaN   
1          325000             21  ...         NaN        NaN          NaN   
2          129000             26  ...         NaN        NaN          NaN   
3          325000             10  ...         NaN        NaN          NaN   
4          325000             10  ...         NaN        NaN          NaN   

  do_nhay  tan_so huong_dan_bao_quan  dac_diem_noi_bat dieu_kien_su_dung  \
0     NaN     NaN                NaN               NaN               NaN   
1     NaN     NaN                NaN               NaN               NaN   
2     NaN     NaN                NaN               NaN               NaN   
3     NaN     NaN                NaN               NaN               NaN   
4     NaN     NaN                NaN               NaN               NaN   

  dia_chi_su_dung bluetooth  
0             NaN       NaN  
1             NaN       NaN  
2             NaN       NaN  
3             NaN       NaN  
4             NaN       NaN  

[5 rows x 84 columns]

***

Drop cột 'Unnamed: 0'

In [28]:
df = df.drop(columns='Unnamed: 0')
df.head()

id  master_id            sku  \
0  204317934  204317934  9972713357587   
1  205279963  205279963  7618382614299   
2  193209826  193209826  5001141446127   
3  205288789  205288789  9035774965791   
4  204649516  204649516  8708257676679   

                                                name  \
0     1111 - Nhật Ký Sáu Vạn Dặm Trên Yên Xe Cà Tàng   
1     1111 - Nhật Ký Sáu Vạn Dặm Trên Yên Xe Cà Tàng   
2  Con Đường Tơ Lụa - Vạn Dặm Xa Từ Pakistan Tới ...   
3     1111 - Nhật Ký Sáu Vạn Dặm Trên Yên Xe Cà Tàng   
4  Sách - 1111 - NHẬT KÝ SÁU VẠN DẶM TRÊN YÊN XE ...   

                                           short_url book_cover   price  \
0  https://tiki.vn/product-p204317934.html?spid=2...        NaN  262000   
1  https://tiki.vn/product-p205279963.html?spid=2...    Bìa mềm  256750   
2  https://tiki.vn/product-p193209826.html?spid=1...    Bìa mềm   95100   
3  https://tiki.vn/product-p205288789.html?spid=2...    Bìa mềm  292500   
4  https://tiki.vn/product-p204649516.html?spid=2...        NaN  292500   

   original_price  discount_rate  rating_average  ...  kich_thuoc tai_trong  \
0          325000             19             5.0  ...         NaN       NaN   
1          325000             21             0.0  ...         NaN       NaN   
2          129000             26             4.8  ...         NaN       NaN   
3          325000             10             0.0  ...         NaN       NaN   
4          325000             10             0.0  ...         NaN       NaN   

  device_brand  do_nhay tan_so  huong_dan_bao_quan dac_diem_noi_bat  \
0          NaN      NaN    NaN                 NaN              NaN   
1          NaN      NaN    NaN                 NaN              NaN   
2          NaN      NaN    NaN                 NaN              NaN   
3          NaN      NaN    NaN                 NaN              NaN   
4          NaN      NaN    NaN                 NaN              NaN   

  dieu_kien_su_dung dia_chi_su_dung bluetooth  
0               NaN             NaN       NaN  
1               NaN             NaN       NaN  
2               NaN             NaN       NaN  
3               NaN             NaN       NaN  
4               NaN             NaN       NaN  

[5 rows x 83 columns]

Xem thử dữ liệu có bị trùng hay không?

In [29]:
df.duplicated().sum()

8294

Vậy là dữ liệu có 8294 dòng trùng nhau, ta sẽ tiến hành xoá những dòng này

In [30]:
df = df.drop_duplicates()
df.shape

(88353, 83)